packages/access-decision-manager-express/README.md

```diff
@@ -0,0 +1,151 @@
+
+
+ # Access-decision-manager-express.
+
+
+ Determines whether or not an entity is entitled to perform a certain action,
+for example, request a certain URI or modify a certain object. It relies on
+[access-decision-manager](https://github.com/wizeline/access-decision-manager/tree/master/packages/access-decision-manager#readme), and to make a positive verdict, it is sufficient to have the approval
+of only one Voter. Voters are implemented by the client service.
+
+
+
+ ### Implementing Voters
+
+ Voters are to determine access rights
+The use of Voters to verify permissions is based on [this model](https://symfony.com/doc/current/security/voters.html).
+Voters are called to grant access to a resource or an action.
+
+
+
+
+  Adding a Custom Voter.
+
+
+  To create a Voter,
+use the `Voter` interface:
+
+ ```typescript
+export interface Voter {
+  supports: (attribute: any, subject: any, context: any) => boolean;
+  voteOnAttribute: (
+    attribute: any,
+    subject: any,
+    user: any,
+    context: any,
+  ) => boolean | Promise<boolean>;
+}
+```
+
+
+ For example, the following voter checks attributes related to a public resources and private ones:
+
+ ```typescript
+import { Voter } from '@wizeline/access-decision-manager';
+
+const supportedAttributes = [
+  'GET_PUBLIC'
+];
+
+const publicVoter = (
+  _options: {}, // eslint-disable-line @typescript-eslint/no-unused-vars
+): Voter => {
+  const supports = (attribute): boolean => {
+    return supportedAttributes.includes(attribute);
+  };
+
+  const voteOnAttribute = (_attribute, subject, user): boolean => {
+    return true;
+  };
+
+  return {
+    supports,
+    voteOnAttribute,
+  };
+};
+
+export default publicVoter;
+
+ ```
+ 
+ 
+ ```typescript
+import { Voter } from '@wizeline/access-decision-manager';
+
+const supportedAttributes = [
+  "GET_PRIVATE"
+];
+
+const publicVoter = (
+  _options: {}, // eslint-disable-line @typescript-eslint/no-unused-vars
+): Voter => {
+  const supports = (attribute): boolean => {
+    return supportedAttributes.includes(attribute);
+  };
+
+  const voteOnAttribute = (_attribute, subject, user): boolean => {
+    // do the logic here to validate determine if has permissions.. maybe
+    // return user.username === 'admin'
+    return false;
+  };
+
+  return {
+    supports,
+    voteOnAttribute,
+  };
+};
+
+export default publicVoter;
+
+```
+
+Then let's add our voters to our express application, here we are assuming that we have the data of our user in req.user
+
+
+```typescript
+import AccessDecisionManagerProvider from '@wizeline/access-decision-manager-express';
+import { Voter } from '@wizeline/access-decision-manager';
+import express from 'express'
+import publicVoter from '../path/to/voters';
+import privateVoter from '../path/to/voters';
+
+const voters = (options): Voter[] => {
```

Comment:
When we similify the voters to not need options this can become:
```typescript
const voters: Voter[] = [
    publicVoter,
    privateVoter
];
```